homeassistant/components/solax/__init__.py

```diff
@@ -1 +1,43 @@
 """The solax component."""
+import logging
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.typing import ConfigType
+
+from .const import DOMAIN
+
+PLATFORMS = ["sensor"]
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the solax component."""
+    if DOMAIN not in config:
+        return True
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=config[DOMAIN]
+        )
+    )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up the sensors from a ConfigEntry."""
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = dict(entry.data)
```

Comment:
```suggestion
    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = dict(entry.data)
```

Why store entry data here?

Reply:
I also based this code from another integration. Should I change this to something else?

Reply:
See suggestion?

Reply:
I understood your suggestion as an optimization to the current code but your PS seems to be asking why I'm storing the data like this. Is there an alternative way besides hass.data?

Reply:
Oh its about the PS part. Sorry misunderstood that.

So, why do you store this info in hass data? And is that needed even?

Reply:
Linking this to the previous comment, so what you're saying is that I can drop the **async_setup** and **async_setup_entry** functions?